symengine/polys/cancel.h

```diff
@@ -0,0 +1,42 @@
+#ifndef SYMENGINE_CANCEL_H
+#define SYMENGINE_CANCEL_H
+
+#include <symengine/basic.h>
+#include <symengine/polys/basic_conversions.h>
+
+using SymEngine::RCP;
+using SymEngine::divides_upoly;
```

Comment:
In header files we should not use `using ...`, because that spills into user code. But since everything below is in the `SymEngine` namespace anyway, I think you can just remove the above two lines.

You can use `using` in `cpp` files because there it won't spill into user code.

Reply:
Okay :+1: nice point.